core/Transaction.go

```diff
@@ -0,0 +1,98 @@
+package core
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+// Todo: define contract class
+type ContractClass struct{}
+
+type DeployTransaction struct {
+	// A random number used to distinguish between different instances of the contract.
+	ContractAddressSalt *felt.Felt
+	// The object that defines the contract’s functionality.
+	ContractDef ContractClass
+	// The arguments passed to the constructor during deployment.
+	ConstructorCalldata []*felt.Felt
+	// Who invoked the deployment. Set to 0 (in future: the deploying account contract).
+	CallerAddress *felt.Felt
+	// The transaction’s version. Possible values are 1 or 0.
+	//
+	// When the fields that comprise a transaction change,
+	// either with the addition of a new field or the removal of an existing field,
+	// then the transaction version increases.
+	// Transaction version 0 is deprecated and will be removed in a future version of StarkNet.
+	version *felt.Felt
```

Comment:
Maybe the version should be a `uint`?
```suggestion
	version uint
```

Reply:
Thanks to @omerfirmak for pointing out that version is a felt in the specs.